src/UI/Components/Diagram/Canvas.tsx

```diff
@@ -1,312 +1,129 @@
-        body: {
-          current_version: instance.instance.version,
-          value: JSON.stringify(coordinates),
-        },
-      });
-    }
-
-    oderMutation.mutate(serviceOrderItems);
-  };
-
-  /**
-   * Handles the update of a service entity block.
-   *
-   * @param {ServiceEntityBlock} cell - The service entity block to be updated.
-   * @param {ActionEnum} action - The action to be performed on the service entity block.
-   */
-  const handleUpdate = (cell: ServiceEntityBlock, action: ActionEnum) => {
-    const newInstance: ComposerServiceOrderItem = {
-      instance_id: cell.id,
-      service_entity: cell.getName(),
-      config: {},
-      action: null,
-      attributes: cell.get("sanitizedAttrs"),
-      edits: null,
-      embeddedTo: cell.get("embeddedTo"),
-      relatedTo: cell.getRelations(),
-    };
-
-    setInstancesToSend((prevInstances) => {
-      const updatedInstance = prevInstances.get(cell.id as string);
-      switch (action) {
-        case "update":
-          newInstance.action =
-            updatedInstance?.action === "create" ? "create" : "update";
-          return new Map(prevInstances.set(cell.id as string, newInstance));
-        case "create":
-          newInstance.action = action;
-          return new Map(prevInstances.set(cell.id as string, newInstance));
-        default:
-          if (
-            updatedInstance &&
-            (updatedInstance.action === null ||
-              updatedInstance.action === "update")
-          ) {
-            return new Map(
-              prevInstances.set(cell.id as string, {
-                instance_id: cell.id,
-                service_entity: cell.getName(),
-                config: {},
-                action: "delete",
-                attributes: null,
-                edits: null,
-                embeddedTo: cell.attributes.embeddedTo,
-                relatedTo: cell.attributes.relatedTo,
-              }),
-            );
-          } else {
-            const newInstances = new Map(prevInstances);
-            newInstances.delete(cell.id as string);
-            return newInstances;
-          }
-      }
-    });
-  };
 
   useEffect(() => {
-    const connectionRules = createConnectionRules(services, {});
+    const connectionRules = createConnectionRules(
+      serviceModels.concat(mainService),
+      {},
+    );
     const actions = diagramInit(
-      canvas,
+      Canvas,
+      (newScroller) => {
+        if (!scroller) {
+          setScroller(newScroller);
+        }
+      },
       connectionRules,
-      handleUpdate,
       editable,
+      mainService,
     );
     setDiagramHandlers(actions);
+    const isMainInstance = true;
+    const newInstances = new Map();
+
     if (instance) {
-      const isMainInstance = true;
-      try {
-        const cells = actions.addInstance(instance, services, isMainInstance);
-        const newInstances = new Map();
+      const cells = actions.addInstance(
+        isMainInstance,
```

Comment:
I believe this has a default already set to true, perhaps put the argument as optional, an at the end of the argument list to avoid needing to create a boolean variable when it's the default :)

Reply:
removed